local_pathfinding/node_mock_global_path.py

```diff
@@ -0,0 +1,449 @@
+
+        Depending on the boolean value of the write parameter, each generated path may be written
+        to a new csv file in the same directory as the source csv file.
+
+        Global path can be changed by modifying mock_global_path.csv or modifying the
+        global_path_filepath parameter.
+
+        Args:
+            gps_call (bool, optional):
+                Whether the callback was called by the gps callback.
+            force (bool, optional):
+                Whether to force the callback to run, without any checks
+        """
+        if not self._all_subs_active():
+            self._log_inactive_subs_warning()
+
+        file_path = self.get_parameter("global_path_filepath")._value
+
+        # check when global path was changed last
+        path_mod_tmstmp = time.ctime(os.path.getmtime(file_path))
+
+        # check if the global path has been forced to update by parameter change in CLI
+        force = self.get_parameter("force")._value
+
+        # Only publish path if the path has changed or gps has changed by more than gps_threshold
+        if (
+            force
+            or gps_call
+            or (self.file_path != file_path)
+            or (path_mod_tmstmp != self.path_mod_tmstmp)
+        ):
+            self.set_parameters([rclpy.Parameter("force", rclpy.Parameter.Type.BOOL, False)])
+
+            self.get_logger().info(
+                f"Global path file is: {os.path.basename(file_path)}\n Reading path"
+            )
+
+            global_path = Path()
+
+            with open(file_path, "r") as file:
+                reader = csv.reader(file)
+                # skip header
+                reader.__next__()
+                for row in reader:
+                    global_path.waypoints.append(
+                        HelperLatLon(latitude=float(row[0]), longitude=float(row[1]))
+                    )
+
+            self.path_mod_tmstmp = path_mod_tmstmp
+            self.file_path = file_path
+            pos = self.gps.lat_lon
+
+            # obtain the actual distances between every waypoint in the global path
+            path_spacing = MockGlobalPath.interval_spacing(pos, global_path.waypoints)
+
+            # obtain desired interval spacing
+            interval_spacing = self.get_parameter("interval_spacing")._value
+
+            # check if global path is just a destination point
+            if len(global_path.waypoints) < 2:
+                self.get_logger().info(
+                    f"Generating new path from {pos.latitude:.4f}, {pos.longitude:.4f} to "
+                    f"{global_path.waypoints[0].latitude:.4f}, "
+                    f"{global_path.waypoints[0].longitude:.4f}"
+                )
+
+                write = self.get_parameter("write")._value
+                if write:
+                    self.get_logger().info("Writing generated path to new file")
+
+                msg = MockGlobalPath.generate_path(
+                    dest=global_path.waypoints[0],
```

Comment:
If you pass in `waypoints` instead of the first element of `waypoints`, then you wouldn't need to handle 2 `dest` types in `generate_path`, right?

Reply:
you are right, I had an idea to make it accept a path as the destination early on, for use in the interpolator, but didn't end up using that. So I will actually get rid of the option for a list arg as its never used.